Framework/DataHandling/test/LoadMuonNexusV2Test.h

```diff
@@ -124,6 +124,37 @@ class LoadMuonNexusV2Test : public CxxTest::TestSuite {
     TS_ASSERT_DELTA(deadTimeTable->Double(62, 1), 0.0073113599, 1e-6);
   }
 
+  void testExecWithTimeZeroTable() {
+    LoadMuonNexusV2 ld;
+    ld.initialize();
+    ld.setPropertyValue("Filename", "EMU00102347.nxs_v2");
+    ld.setPropertyValue("OutputWorkspace", "outWS");
+    ld.setPropertyValue("TimeZeroTable", "tzt");
+
+    TS_ASSERT_THROWS_NOTHING(ld.execute());
```

Comment:
You need to do `ld.setRethrows(true);` before calling execute, otherwise this will always pass

Personally I would do something like this, as it reduces the boilerplate whilst doing the exact same thing:
```
ld.setRethrows(true);
// An unhandled exception here will cause the thread to terminate with the exception, rather than risk TS_ASSERT_THROWS_NOTHING swallowing / losing some info 
ld.execute()

auto &adsInst = ADS::Instance();  // As you reuse it later in the test too

TS_ASSERT(adsInst.doesExist("outWS"))
Workspace2D_sptr output2D = ....
...
```

Reply:
Thanks David that's a really helpful explanation. I think at the time I just tied to copy what other tests were doing without giving it enough thought of my own

Reply:
Not at all, I had similar feedback myself when I was a student too. Plus less boilerplate means you can spend more time solving interesting problems instead too :)